util/util.go

```diff
@@ -383,3 +383,18 @@ func HasOwner(refList []metav1.OwnerReference, apiVersion string, kinds []string
 
 	return false
 }
+
+// SplitMachineList separates the machines running the control plane from other worker nodes.
+func SplitMachineList(list *clusterv1.MachineList) (*clusterv1.MachineList, *clusterv1.MachineList) {
```

Comment:
This is very specific and I'd prefer not to have it in util. Could you move it into cluster_controller.go?

Reply:
sgtm